js/coinsbit.js

```diff
@@ -0,0 +1,682 @@
+'use strict';
+
+// ---------------------------------------------------------------------------
+
+const Exchange = require ('./base/Exchange');
+const { ExchangeError } = require ('./base/errors'); // , BadRequest, AuthenticationError, RateLimitExceeded, BadSymbol, InvalidOrder, InsufficientFunds, ArgumentsRequired, OrderNotFound, PermissionDenied
+const { TICK_SIZE } = require ('./base/functions/number');
+const Precise = require ('./base/Precise');
+// function c(o){console.log(o);}
+// ---------------------------------------------------------------------------
+
+module.exports = class coinsbit extends Exchange {
+    describe () {
+        return this.deepExtend (super.describe (), {
+            'id': 'coinsbit',
+            'name': 'Coinsbit',
+            'countries': [ 'EE' ], // Estonia
+            // 'rateLimit': 50, // default rate limit is 20 times per second
+            'version': 1,
+            'certified': false,
+            'has': {
+                'addMargin': undefined,
+                'cancelAllOrders': undefined,
+                'cancelOrder': undefined,
+                'createMarketOrder': undefined,
+                'createOrder': undefined,
+                'fetchBalance': undefined,
+                'fetchCanceledOrders': undefined,
+                'fetchClosedOrders': undefined,
+                // 'fetchCurrencies' doesn't exist, and will be 'emulated'
+                'fetchDepositAddress': undefined,
+                'fetchDeposits': undefined,
+                'fetchFundingRateHistory': undefined,
+                'fetchMarkets': true,
+                // 'fetchMarketsByType': <-- i think fetchMarketsByType makes things complex, and might be totally removed from ccxt. There should be one 'fetchMarkets' imho. you can see another comment related to this matter under 'checkExchangeType' function down below.
+                'fetchMyTrades': undefined,
+                'fetchOHLCV': true,
+                'fetchOpenOrders': undefined,
+                'fetchOrder': undefined,
+                'fetchOrderBook': true,
+                'fetchOrderTrades': undefined,
+                'fetchPosition': undefined,
+                'fetchPositions': undefined,
+                'fetchStatus': 'emulated',
+                'fetchTicker': true,
+                'fetchTickers': true,
+                'fetchTime': 'emulated',
+                'fetchTrades': undefined,
+                'fetchWithdrawals': undefined,
+                'reduceMargin': undefined,
+                'setLeverage': undefined,
+                'withdraw': undefined,
```

Comment:
I suggest removing all has properties from here and adding only those that are 100%-true or 100%-false.

Reply:
That's sure, in final submission (after I do PRIVATE part too), I'll be removing all unneeded ones. atm, during development, I've just pasted there all possible ones to easily (not going to other exchanges and copying) set them true/false.

Reply:
Ok, it would just help keep track of what has been actually added to the moment in this PR.